src/attribution/onTTFB.ts

```diff
@@ -24,10 +24,22 @@ import {
 } from '../types.js';
 
 const attributeTTFB = (metric: TTFBMetric): void => {
-  if (metric.entries.length) {
+  // Only attribute if you have a non-zero responseStart value and a
+  // navigationEntry entry.
+  // E.g. For cross-origin redirects, Safari has a workerStart time but no
+  // TTFB time, so do not provide attribution for that.
+  if (metric.entries.length && metric.entries[0].responseStart) {
     const navigationEntry = metric.entries[0];
     const activationStart = navigationEntry.activationStart || 0;
 
+    // Measure from workerStart or fetchStart so any service worker startup
+    // time is included in cacheDuration (which also includes other sw time
+    // anyway, that cannot be accurately split out cross-browser).
+    const waitEnd = Math.max(
+      (navigationEntry.workerStart || navigationEntry.fetchStart) -
```

Comment:
Maybe it makes more sense to put the SW startup time in "waiting" WDYT?

Reply:
As per the comment above this, I preferred it in `cacheDuration` as then get a better redirect time in `waitingDuration`.

Also all the other service worker work is in `cacheDuration` so nice to haver them together (given we can't separate them out cleanly as discussed offline).

But I do agree it's technically "waiting time".